examples/rococo/assetHub/foreignAssetTransfers/reserve/assetHubMUSEToBifrostRemoteReserve.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * When importing from @substrate/asset-transfer-api it would look like the following
+ *
+ * import { AssetTransferApi, constructApiPromise } from '@substrate/asset-transfer-api'
+ */
+import { AssetTransferApi, constructApiPromise } from '../../../../../src';
+import { TxResult } from '../../../../../src/types';
+import { GREEN, PURPLE, RESET } from '../../../../colors';
+
+/**
+ * In this example we are creating a `polkadotXcm` pallet `transferAssetsUsingTypeAndThen` call to send MUSE (foreign asset with location `{"parents":"1","interior":{"X1":{"Parachain":"3369"}}}`)
+ * from a Rococo Asset Hub (System Parachain) account
+ * to a Bifrost Polkadot (Parachain) testnet account, where the `xcmVersion` is set to 4, and there is no
+ * `weightLimit` option provided which declares that the tx will allow unlimited weight to be used for fees.
+ * We provide a `paysWithFeeDest` asset location for MUSE which will be used to pay for fees on the destination chain.
+ * We also provide a value of `RemoteReserve` for the `assetTransferType` and `feeTransferType` options declaring a custom reserve location for both
+ * the `remoteReserveAssetTransferTypeLocation` and `remoteReserveFeesTransferTypeLocation` options.
+ *
+ * NOTE: To specify the amount of weight for the tx to use provide a `weightLimit` option containing desired values for `refTime` and `proofSize`.
+ */
+const main = async () => {
+	const { api, specName, chainName, safeXcmVersion } = await constructApiPromise(
+		'wss://rococo-asset-hub-rpc.polkadot.io',
+	);
+	const assetApi = new AssetTransferApi(api, specName, safeXcmVersion, {
+		chainName,
+	});
+
+	let callInfo: TxResult<'call'>;
+	try {
+		callInfo = await assetApi.createTransferTransaction(
+			`2030`,
+			'0x6E733286C3Dc52C67b8DAdFDd634eD9c3Fb05B5B',
+			[`{"parents":"1","interior":{"X1":{"Parachain":"3369"}}}`],
+			['1000000000000'],
+			{
+				format: 'call',
+				xcmVersion: 4,
+				paysWithFeeDest: '{"parents":"1","interior":{"X1":{"Parachain":"3369"}}}',
+				assetTransferType: 'RemoteReserve',
+				remoteReserveAssetTransferTypeLocation: '{"parents":"1","interior":{"X1":{"Parachain":"3369"}}}',
```

Comment:
When the remote reserve used is the same as asset "origin/id", then `pallet_xcm::transferAssets` also works, no need to use the more complicated `transferAssetsUsingTypeAndThen`.

`transferAssetsUsingTypeAndThen` is required when using "other" reserves (such as AssetHub) or when needing to do something else with the asset once it arrives (specify xcm to run on final dest).

Reply:
Probably this example was meant to show how to do a remote reserve transfer using the new `transferAssetsUsingTypeAndThen`, but I am ok with simplifying it with the "default case". 

Also, as commented [here](https://github.com/paritytech/asset-transfer-api/issues/400#issuecomment-2102235785), this PR gives initial support to the new call. We should add meaningful examples for the complex use cases once they are nicely supported by the API.

Reply:
yes, I don't mean you should change it, just wanted to share the information :)